defensics/tooling.py

```diff
@@ -0,0 +1,181 @@
+    def __init__(self, testcase=None, port='/dev/ttyACM0', bauderate=115200,
+                 path=os.getcwd(), newtmgr=None):
+        self.connection = serial.Serial(port=port, baudrate=bauderate, timeout=None)
+        self.testcase = testcase
+        self.path = path
+        self.process = None
+        self.shutdown = False
+        self.newtmgr = newtmgr
+
+        super(SerialOutput, self).__init__()
+
+    def run(self):
+        logging.debug('Capturing console output')
+        if self.testcase:
+            filename = str(self.testcase) + '.txt'
+        else:
+            filename = 'test.txt'
+        file = open(filename, 'a')
+        while not self.shutdown:
+            # limit data acquisition frequency
+            time.sleep(1)
+            bytesToRead = self.connection.inWaiting()
+            data = self.connection.read(bytesToRead).decode('latin-1')
+            if crash_detection:
+                if 'Unhandled interrupt' in data:
+                    self.newtmgr.check_corefile()
+            file.write(data)
+            file.flush()
+        # before ending wait half a second for data to appear; save what arrived
+        time.sleep(1)
+        bytesToRead = self.connection.inWaiting()
+        data = self.connection.readline(bytesToRead).decode()
+        logging.debug('Saving console output file')
+        file.write(data)
+        file.close()
+
+
+class NewtMgr:
+    def __init__(self, profile_name='test', type='oic_ble', connstring='peer_name=c5', testcase=None):
+        self.profile_name = profile_name
+        self.type = type
+        self.connstring = connstring
+        self.testcase = testcase
+
+    def make_profile(self):
+        """
+        This method adds new connection profile or overwrites the old one, if it exists
+        """
+        cmd = ['sudo', '-S', 'newtmgr', 'conn', 'add', self.profile_name,
+               'type=' + self.type, 'connstring=' + self.connstring]
+        process = subprocess.Popen(cmd, shell=False, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+        output = process.communicate(input=pwd.encode())[0].decode()
+        logging.debug(output)
+        ctrl = 'Connection profile ' + self.profile_name + ' successfully added'
+        if ctrl in output:
+            logging.debug('Connection profile added')
+        else:
+            logging.debug('Failed to add connection profile ')
+
+    def check_corefile(self):
+        """
+        Check if corefile exists; if board is not responding, restart it and check again
+        """
+        cmd = ['sudo', '-S', 'newtmgr', 'image', 'corelist', '-c', self.profile_name]
+        process = subprocess.Popen(cmd, shell=False, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+        output = process.communicate(input=pwd.encode())[0].decode()
+        logging.debug(output)
+        if 'Corefile present' in output:
+            logging.debug('Board has crashed, corefile present')
+            self.download_and_delete_corefile()
```

Comment:
We are not handling errors here.

Reply:
As errors here are not very dangerous, they're only being logged now